providers/providerverifier.go

```diff
@@ -132,6 +132,11 @@ func (v *DefaultProviderVerifier) VerifyIDToken(ctx context.Context, idToken []b
 			return fmt.Errorf("failed to get OP public key: %w", err)
 		}
 
+		_, ok := pubKeyRecord.PublicKey.(*rsa.PublicKey)
```

Comment:
Nit: added a clarifying comment here 
```suggestion
        // Ensure that the algorithm of public key from OpenID Provider matches 
        // the algorithm specified in the ID Token
		_, ok := pubKeyRecord.PublicKey.(*rsa.PublicKey)
```